torch/_dynamo/guards.py

```diff
@@ -2188,7 +2200,7 @@ def __init__(
         w_builder = None
 
         self.torch_function_mode_stack = (
-            output_graph.torch_function_mode_stack if output_graph else None
+            torch.overrides._get_current_function_mode_stack()
```

Comment:
Won't this always return empty since we clear the stack inside compile node via `preserve_global_state`

Reply:
No, we've already returned from `transform` by the time we call the check function manager so the stack is restored by then.